### PR TITLE
logseq: make dugite use the `git` in nixpkgs

### PR DESCRIPTION
**Pull request by SaltyKitkat**
###### Description of changes

Make logseq use the git in nixpkgs so that it will benefit from security patches and so on in git. Also, there will be no more dynamic linking problems like #167967.

Remove the git binary delivered with `logseq`, since it's not necessary anymore.

Thanks @qbit for the inspiration!

This has been test on my own machine and both git (auto)commit and git push seems working very well.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SaltyKitkat** [comment]
This will close #167967

**Comment by qbit** [comment]
Result of `nixpkgs-review pr 177704` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>logseq</li>
  </ul>
</details>

File: pkgs/applications/misc/logseq/default.nix [hunk 1]
```diff
@@ -29,6 +28,12 @@ stdenv.mkDerivation rec {
     cp -a ${appimageContents}/{locales,resources} $out/share/${pname}
     cp -a ${appimageContents}/Logseq.desktop $out/share/applications/${pname}.desktop
 
+    # remove the `git` in `dugite` because we want the `git` in `nixpkgs`
+    chmod +w -R $out/share/${pname}/resources/app/node_modules/dugite/git
+    chmod +w $out/share/${pname}/resources/app/node_modules/dugite
+    rm -rf $out/share/${pname}/resources/app/node_modules/dugite/git
+    chmod -w $out/share/${pname}/resources/app/node_modules/dugite
```
Comment (marsam) [inline pkgs/applications/misc/logseq/default.nix:35]: I think it's not necessary to change the permissions:
```suggestion
    rm -rf $out/share/${pname}/resources/app/node_modules/dugite/git
```
besides that, LGTM.

Comment (SaltyKitkat) [reply pkgs/applications/misc/logseq/default.nix:35]: It seems that these files are extracted from appimage and with permission `r-xr-xr-x`. So it's necessary to set the write permission to remove the `git` directory.

I'm not familiar with `nix` and I get the permission info as following:

```
$ ls -l $out/share/logseq/resources/app/node_modules
total 0
dr-xr-xr-x 1 nixbld1 nixbld   0 Jan  1  1970 '@develar'/
dr-xr-xr-x 1 nixbld1 nixbld   6 Jan  1  1970 '@electron'/
dr-xr-xr-x 1 nixbld1 nixbld   0 Jan  1  1970 '@electron-forge'/
......
dr-xr-xr-x 1 nixbld1 nixbld 230 Jan  1  1970  dugite/
......
```

Comment (SaltyKitkat) [reply pkgs/applications/misc/logseq/default.nix:35]: What they do is not exactly the same.

`chmod +w -R $out/share/${pname}/resources/app/node_modules/dugite`
will recurse into other dirs besides `git`.

But maybe it's not a big problem... I don't know

**Pull request merged**